### ISSUE TITLE
Sender into_sink() panics when used with futures forward.

### ISSUE THREAD

**Issue by stsydow** [zesterer/flume#55]
A sender sink created with into_sink() panics when used with futures stream::forward().
Using it in an async closure with fold works fine still. (see out commended code below)

https://github.com/zesterer/flume/blob/b37cc6bfddf588b34c7c82a52531216a26b7eb00/src/async.rs#L146

Trace:

> thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/flume-0.9.1/src/async.rs:146:40
stack backtrace:
   0: rust_begin_unwind
             at /rustc/dd7fc54ebdca419ad9d3ab1e9f5ed14e770768ea/library/std/src/panicking.rs:483:5
   1: core::panicking::panic_fmt
             at /rustc/dd7fc54ebdca419ad9d3ab1e9f5ed14e770768ea/library/core/src/panicking.rs:85:14
   2: core::panicking::panic
             at /rustc/dd7fc54ebdca419ad9d3ab1e9f5ed14e770768ea/library/core/src/panicking.rs:50:5
   3: core::option::Option<T>::unwrap
             at /root/.rustup/toolchains/nightly-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:386:21
   4: <flume::async::SendFuture<T> as core::future::future::Future>::poll
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/flume-0.9.1/src/async.rs:146:23
   5: <flume::async::SendSink<T> as futures_sink::Sink<T>>::poll_ready
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/flume-0.9.1/src/async.rs:187:9
   6: <futures_util::sink::map_err::SinkMapErr<Si,F> as futures_sink::Sink<Item>>::poll_ready
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.6/src/sink/map_err.rs:39:9
   7: <futures_util::stream::stream::forward::Forward<St,Si,Item> as core::future::future::Future>::poll
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.6/src/stream/stream/forward.rs:59:24
   8: <futures_util::stream::stream::Forward<St,Si> as core::future::future::Future>::poll
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.6/src/lib.rs:111:13
   9: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.0/src/runtime/task/core.rs:173:17
  10: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.0/src/loom/std/unsafe_cell.rs:14:9
  11: tokio::runtime::task::core::Core<T,S>::poll
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.0/src/runtime/task/core.rs:158:13
  12: tokio::runtime::task::harness::Harness<T,S>::poll::{{closure}}
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.0/src/runtime/task/harness.rs:107:27

Test code: (excerpt form my test suite)
(build with tokio v0.3 released recently)

```rust
use tokio::runtime::Runtime;
use futures::future::{self, FutureExt};
use futures::stream::{self, Stream, StreamExt};
  
use test::Bencher;


const BLOCK_COUNT:usize = 1_000;
  
const THREADS:usize = 8;
  
const CHANNEL_BUFFER:usize = 100;

 type Message = usize;
  fn new_message(val:usize) -> Message {
      val
  }
  
  fn dummy_stream() -> impl Stream<Item=Message> {
      stream::unfold(0, |count| async move {
          if count < BLOCK_COUNT {
              let buffer = new_message(count);
              Some((buffer, count + 1))
          } else {
              None
          }
      })
  }
  
#[bench]
fn channel_buf100_flume(b: &mut Bencher) {
    let runtime = Runtime::new().expect("can not start runtime");
    b.iter(|| {
        let (send_task, recv_task) = {
        use futures::SinkExt;
        let (tx, rx) = flume::bounded::<Message>(100);

        /*
        let send_task = dummy_stream()
            .fold(tx, |mut tx, item| async move {
                let r = tx.send_async(item).await;
                r.expect("Receiver closed");
                tx
            });
        */
        let send_task = dummy_stream()
            .map(|i| Ok(i))
            .forward(tx.into_sink().sink_map_err(|e| {
                panic!("send error:{:#?}", e)
            }));

        let recv_task = rx
            .into_stream()
            .for_each(|item| async move {
                test_msg(item);
            });
        (send_task, recv_task)
        };
        runtime.spawn(send_task);
        runtime.block_on(recv_task);
    });
}
```

**Comment by Restioson** [comment]
I've got a fix in #56.

**Comment by Restioson** [comment]
Merged.
